src/mod/server.mod/servmsg.c

```diff
@@ -1734,6 +1745,16 @@ static int gotcap(char *from, char *msg) {
   return 1;
 }
 
+static int server_isupport(char *key, char *isset_str, char *value)
+{
+  int isset = !strcmp(isset_str, "1");
+
+  if (!strcmp(key, "NICKLEN")) {
+    isupport_parseint(key, isset ? value : NULL, 9, NICKMAX, 1, 9, &nick_len);
```

Comment:
Should the second '9' instead be 32, since to match the 32 we specify in isupport_default ?

```
static const char isupport_default[4096] = "CASEMAPPING=rfc1459 CHANNELLEN=80 NICKLEN=32 CHANTYPES=#& PREFIX=(ov)@+ CHANMODES=b,k,l,imnpst MODES=3 MAXCHANNELS=10 TOPICLEN=250 KICKLEN=250 STATUSMSG=@+";
```

Reply:
Assumptions for a scenario:
- the default value 32 is used (server doesn't provide one)
- the server actually uses a maximum nick length of 9
- we add code to this PR that changes the Tcl variable nicklen to 32 (the default value based on above conditions)
- the bot has a 30 character nickname in the config file that, when truncated by the server, matches an already in-use nickname and is rejected by the server, as is the altnick

So now, with these conditions, Eggdrop will try to find a 'random' nickname by appending a random character at the end of the nick, at the 30th character. However, with nicklen being 9 on this network, the server only accepts the first 9 characters of the nick and keeps throwing errors because Eggdrop appears to be using the same nick each connect.

So, is it better to set a default to a lower number, or not try and overwrite nick-len?

Reply:
Set default to 9 to follow eggdrop standards (reality is that it depends on net-type but this will do)